src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewFilterMapper.xml

```diff
@@ -105,6 +105,121 @@
+                        ),
+                        mutated_samples AS (
+                            SELECT DISTINCT sample_unique_id
+                            FROM genomic_event_derived
+                            WHERE cancer_study_identifier IN
+                                <foreach item="studyId" collection="studyViewFilter.studyIds" open="(" separator="," close=")">
+                                    #{studyId}
+                                </foreach>
+                                AND hugo_gene_symbol = #{mutationDataFilter.hugoGeneSymbol}
+                                AND variant_type = 'mutation'
+                        )
+                        SELECT DISTINCT sample_unique_id
+                        FROM (
+                            <foreach item="dataFilterValue" collection="mutationDataFilter.values[0]" separator="UNION ALL">
+                                <if test="dataFilterValue.value == 'MUTATED'">
+                                    SELECT sample_unique_id FROM mutated_samples
+                                </if>
+                                <if test="dataFilterValue.value == 'NOT_MUTATED'">
+                                    SELECT sample_unique_id FROM profiled_samples
+                                    WHERE sample_unique_id NOT IN (SELECT sample_unique_id FROM mutated_samples)
+                                </if>
+                                <if test="dataFilterValue.value == 'NOT_PROFILED'">
+                                    SELECT sample_unique_id FROM total_samples
+                                    WHERE sample_unique_id NOT IN (SELECT sample_unique_id FROM profiled_samples)
+                                </if>
+                            </foreach>
+                        )
+                    </if>
+                    <!-- if the categorization is 'MUTATION_TYPE' (table) -->
+                    <if test="mutationDataFilter.categorization == @org.cbioportal.web.parameter.MutationOption@MUTATION_TYPE">
+                        <foreach item="dataFilterValues" collection="mutationDataFilter.values" separator="INTERSECT">
+                            SELECT DISTINCT sample_unique_id
+                            FROM genomic_event_derived
+                            WHERE hugo_gene_symbol = #{mutationDataFilter.hugoGeneSymbol}
+                                AND variant_type = 'mutation'
+                                AND mutation_type IN (
+                                <foreach item="dataFilterValue" collection="dataFilterValues" separator=",">
+                                    #{dataFilterValue.value}
+                                </foreach>
+                                )
+                        </foreach>
+                    </if>
+                </foreach>
+            </if>
+            <if test="studyViewFilter.genomicDataFilters != null and !studyViewFilter.genomicDataFilters.isEmpty()">
```

Comment:
even though we acknowledged that CURRENTLY, genomic-data-counts only handles CNA data, I think we should structure/name our code so that in the future it will be obvious how to add support for new types.  this could mean putting a condition here that detects CNA type in some way.   also comments.  at least then a future reader doesn't puzzle over why we suddenly assume CNA here.

Reply:
Okay sure. I'll add the condition and comments